p2p/protocol/identify/obsaddr.go

```diff
@@ -534,6 +527,22 @@ func (oas *ObservedAddrManager) emitSpecificNATType(addrs []*observedAddr, proto
 	return false, 0
 }
 
+func (oas *ObservedAddrManager) Close() error {
+	oas.closeOnce.Do(func() {
+		oas.ctxCancel()
+
+		oas.mu.Lock()
+		oas.refreshTimer.Stop()
+		oas.refreshTimer = nil
```

Comment:
This doesn't work as the `worker` might be blocked on the timer channel, creating a race condition.

Reply:
Eh, we could have a "shutdown" boolean. That or have the worker keep a copy of the timer and/or channel locally instead of reading `oas.refreshTimer`.